src/JWT/JwtValidator.cs

```diff
@@ -71,10 +71,10 @@ public JwtValidator(IJsonSerializer jsonSerializer, IDateTimeProvider dateTimePr
         /// <param name="validationParameters">Validation parameters that are passed on to <see cref="JwtValidator"/></param>
         public JwtValidator(IJsonSerializer jsonSerializer, IDateTimeProvider dateTimeProvider, int timeMargin, ValidationParameters valParams)
         {
-            _jsonSerializer = jsonSerializer;
-            _dateTimeProvider = dateTimeProvider;
+            _jsonSerializer = jsonSerializer ?? throw new ArgumentNullException(nameof(jsonSerializer));
+            _dateTimeProvider = dateTimeProvider ?? throw new ArgumentNullException(nameof(dateTimeProvider));

```

Comment:
The tests in `JwtValidatorTests` are failing due to this change; they are expecting an exception of type `ArgumentException` not `ArgumentNullException`.

I'm not sure if this is enough to be considered a breaking change (in which case the library version would have to be bumped to `9.0.0`).

Reply:
Oh, wow. Can you please update the tests, if not already?

And yes, I think it's fine to bump the version to 9.0.0, did that.

Reply:
Tests have already been updated; they are all passing.